src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -173,17 +183,23 @@ private void removeDeadDataNodes() {
     return allTasksIds.values();
   }
 
-  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType) {
+  public void addHdfsNode(Protos.TaskID taskId, String hostname, String taskType, String taskName) {
     switch (taskType) {
       case HDFSConstants.NAME_NODE_ID :
         HashMap<String, String> nameNodes = getNameNodes();
         nameNodes.put(hostname, taskId.getValue());
         setNameNodes(nameNodes);
+        HashMap<String, String> nameNodeTaskNames = getNameNodeTaskNames();
+        nameNodeTaskNames.put(taskId.getValue(), taskName);
```

Comment:
Should we verify that there is no preexisting entry for that taskId? Otherwise, we would just overwrite the value. Probably won't happen, but worth an assertion check.